opentelemetry-appender-tracing/src/layer.rs

```diff
@@ -180,6 +180,26 @@ where
         // Visit fields.
         event.record(&mut visitor);
 
+        #[cfg(feature = "tracing")]
```

Comment:
@dylanahsmith Really appreciate this work! I need to request changes to address below:

Without resolving https://github.com/open-telemetry/opentelemetry-rust/issues/1571 I am not okay with this change, which recognizes `tracing` as a supported Distributed Tracing API, adds dependency to `tracing-opentelemetry` crate which may not be relevant in the future, and enable it by default.

I can totally see the demand for this (and certainly something we want to do based on outcome of above linked issue!!), so how about the following:
1. Rename the feature-flag to something indicating it is experimental like "experimental-use-tracing-span-context"/similar.
2. Do not enable it by default.

This crate is about to go RC and stable soon, so it is important to ensure everything we put in there are very solid.

@lalitb @bantonsson @scottgerring @TommyCpp Please share your opinions.

Reply:
I agree on adding it as experimental opt-in feature, with `tracing-opentelemetry` as optional dependency.

Reply:
Yup. Agree with @lalitb. It should be experimental opt-in for now.

Reply:
It seems strange to support the `tracing` crate for logs, but not for **tracing**, but makes sense if you are considering providing a replacement for tracing-opentelemetry.

Anyways, I totally get being hesitant to stabilize things too early and have made the requested change to make the feature experimental